### PR TITLE
Turingcup

### PR DESCRIPTION
**Pull request by SalmanAd01**
## Description of the change
Added the webpage for Turing cup with backend integration

## Type Of Change

- [ ] Bug Fix (non-breaking change that fixes an issue)
- [x] New Feature (non-breaking change that adds functionality)
- [ ] Breaking Change (fix or feature that would cause existing functionality to not work as expected)

## Development & Code Review

- [x] Fixed all the formatting errors (run `npm run format`)
- [x] The code changed/added has been covered with tests
- [x] All the test cases added/changed are passing

### CODE REVIEW

**Review by Aman-Pinjara — APPROVED** [review]
LGTM

File: src/Components/turingcup/Registration.jsx [hunk 1]
```diff
@@ -0,0 +1,620 @@
...
+      }
+      if (!branch.includes(data.branchName)) {
+        alert('Branch is not valid');
+        return;
+      }
+      if (!year.includes(data.currentYear)) {
+        alert('Year is not valid');
+        return;
+      }
+      const URL = 'http://localhost:3000/turingcup/register';
```
Comment (ac030540) [inline src/Components/turingcup/Registration.jsx:133]: @SalmanAd01 I think you forgot to update API url here

File: src/Components/turingcup/Navbar.jsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    }
+  };
+  return (
+    <>
+      <link
+        href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css"
+        rel="stylesheet"
+        integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm"
+        crossOrigin="anonymous"
+      />
```
Comment (ac030540) [inline src/Components/turingcup/Navbar.jsx:44]: Why we have opted to add bootstrap this way?

**Review by ac030540 — COMMENTED** [review]
Great work @SalmanAd01 & @Aman-Pinjara :) . Sorry for the late review. I just wanted to highlight the folder structure of our project with you all.

Folder Structure: `src/moduleName/routeName`

In case of turingcup it should be `src/drawer/turingCup`. Since turingCup is present the AppDrawer.

File: src/Components/turingcup/Registration.jsx (hunk 1 above)
Comment (SalmanAd01) [reply src/Components/turingcup/Registration.jsx:133]: I've fix it Thanks

File: src/Components/turingcup/Navbar.jsx (hunk 2 above)
Comment (SalmanAd01) [reply src/Components/turingcup/Navbar.jsx:44]: I'll import in Style.css

**Comment by SalmanAd01** [comment]
> Great work @SalmanAd01 & @Aman-Pinjara :) . Sorry for the late review. I just wanted to highlight the folder structure of our project with you all.
> 
> Folder Structure: `src/moduleName/routeName`
> 
> In case of turingcup it should be `src/drawer/turingCup`. Since turingCup is present the AppDrawer.

I'll open new PR

**Pull request merged**